### PR TITLE
feat: added refs to link components

### PR DESCRIPTION
**Pull request by jedimonkey**
Adds ref support to the nextjs/react <Link> component.

## Description / Motivation
Currently if you are using this library with another library that tries to pass a ref to the Link component, you get warnings as it doesn't support Link's being passed through.
We have been using this with the MUI framework, and they require refs when passing the component to it's components:
https://mui.com/material-ui/guides/composition/
@pzi has opened an issue: https://github.com/Sitecore/jss/issues/1053

## Testing Details
Added to the Image.test.tsx files a simple check that the ref's id is the same as the anchor's id.
Just ran the standard `yarn run test` in both react/nextjs packages and then `yarn run coverage-packages`. Linting passes, however can't build. But then dev wouldn't build for me neither.

## Types of changes
- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

could be seen as a bug, or an enhancement.

### CODE REVIEW

File: packages/sitecore-jss-nextjs/src/components/Link.test.tsx [hunk 1]
```diff
@@ -278,4 +278,21 @@ describe('<Link />', () => {
     expect(rendered.find(NextLink).length).to.equal(0);
     expect(rendered.find(ReactLink).length).to.equal(1);
   });
+
+  it.only('should render with a ref to the anchor', () => {
```
Comment (pzi) [inline packages/sitecore-jss-nextjs/src/components/Link.test.tsx:282]: Probably want to remove the `only` :)

File: packages/sitecore-jss-react/src/components/Link.test.tsx [hunk 2]
```diff
@@ -98,4 +98,17 @@ describe('<Link />', () => {
     const rendered = mount(<Link field={field} id="my-link" />);
     expect(rendered.html()).to.contain('id="my-link"');
   });
+
+  it.only('should render with a ref to the anchor', () => {
```
Comment (pzi) [inline packages/sitecore-jss-react/src/components/Link.test.tsx:102]: same here

File: packages/sitecore-jss-nextjs/src/components/Link.test.tsx (hunk 1 above)
Comment (jedimonkey) [reply packages/sitecore-jss-nextjs/src/components/Link.test.tsx]: 🙃

File: packages/sitecore-jss-react/src/components/Link.tsx [hunk 3]
```diff
@@ -38,88 +38,92 @@ export type LinkProps = React.DetailedHTMLProps<
...
-  return <React.Fragment>{resultTags}</React.Fragment>;
-};
+);
 
 export const LinkPropTypes = {
   field: PropTypes.oneOfType([
     PropTypes.shape({
       href: PropTypes.string,
     }),
+    PropTypes.shape({
```
Comment (jedimonkey) [inline packages/sitecore-jss-react/src/components/Link.tsx:124]: Needed to make this optional due to React's forwardRef parsing this into a WeakValidationMap which makes the field's url null. Happy to change this, but no idea on how to make this use a stricter type. :(

**Comment by jedimonkey** [comment]
I've just updated this as there was a conflict due to a recent merged change. However I noticed you have bumped to react 18. Is it possible if this gets merged we can back port this to the v20 release as this is an issue for us. Not sure how quick it will be for us to move to v18 of react (or if this will be compatible with v17 of react).

**Comment by pzi** [comment]
@ambrauer @illiakovalenko could we get a review/some feedback please?

**Comment by jedimonkey** [comment]
Is there anyone who can have a look into this? The actual change is reasonably small, just the formatting has made it look large. Keen to get this into our project as we have recently upgrade using MUI which is causing lots of warnings for the Link component.

**Comment by ambrauer** [comment]
Hey @jedimonkey @pzi - I've added this to our backlog so we can prioritize getting you some feedback and merging this in. Expect something in the next couple weeks. We are actively planning a minor release of v20 and hoping to include this one.

**Comment by pzi** [comment]
awesome, thanks for getting back to us @ambrauer 👍🏼

**Comment by illiakovalenko** [comment]
@pzi @jedimonkey I pulled your commits into another branch (#1080) in order to make quick changes:
1. Removed package.lock
2. Removed `children` proptype - there is a mismatch which I couldn't handle when you provide different values
We will release it in the next major release as it's recommended in React docs, don't release a minor version if you add Ref's

**Pull request closed without merging**